src/librustdoc/html/static/settings.js

```diff
@@ -2,29 +2,54 @@
 /* global getCurrentValue, updateLocalStorage */
 
 (function () {
-    function changeSetting(settingName, isEnabled) {
-        updateLocalStorage('rustdoc-' + settingName, isEnabled);
+    function changeSetting(settingName, value) {
+        updateLocalStorage('rustdoc-' + settingName, value);
+
+        switch (settingName) {
+            case 'preferred-dark-theme':
+            case 'preferred-light-theme':
+            case 'use-system-theme':
+                updateSystemTheme();
+                break;
+        }
     }
 
     function getSettingValue(settingName) {
         return getCurrentValue('rustdoc-' + settingName);
     }
 
     function setEvents() {
-        var elems = document.getElementsByClassName("slider");
-        if (!elems || elems.length === 0) {
-            return;
+        var elems = {
+            toggles: document.getElementsByClassName("slider"),
+            selects: document.getElementsByClassName("select-wrapper")
+        };
+
+        if (elems.toggles && elems.toggles.length > 0) {
+            for (var i = 0; i < elems.toggles.length; ++i) {
+                var toggle = elems.toggles[i].previousElementSibling;
+                var settingId = toggle.id;
+                var settingValue = getSettingValue(settingId);
+                if (settingValue !== null) {
+                    toggle.checked = settingValue === "true";
+                }
+                toggle.onchange = function() {
+                    changeSetting(this.id, this.checked);
+                };
+            }
         }
-        for (var i = 0; i < elems.length; ++i) {
-            var toggle = elems[i].previousElementSibling;
-            var settingId = toggle.id;
-            var settingValue = getSettingValue(settingId);
-            if (settingValue !== null) {
-                toggle.checked = settingValue === "true";
+
+        if (elems.selects && elems.selects.length > 0) {
+            for (var i = 0; i < elems.selects.length; ++i) {
```

Comment:
`i` has already been declared in the previous loop (in case the previous loop was run), so please create the `i` variable on its own before the two loops (`var i;`).

Reply:
The variables are scoped to the loop anyway. I created the variable on its own like you asked but may I know why it's preferred?

Reply:
I don't think they *are* scoped to the loop - javascript has variable hoisting.

Reply:
```
(bash@joshua-ThinkPad-P51s) ~ ▶️ node
Welcome to Node.js v12.16.1.
Type ".help" for more information.
>  for (var i = 0; i < 5; ++i) {}
undefined
> i
5
```

Reply:
Oh my god. My bad. I got used to using `let` and forgot the semantics of plain-old `var`iables...

Reply:
The "surprises" of javascript. Don't worry, we all went through this. ;)